docs/investigation/multi-cluster/README.md

```diff
@@ -0,0 +1,279 @@
+
+```yaml
+kind: Implementation
+metadata:
+  # ...
+  displayName: Install PostgreSQL database
+spec:
+  # ...
+  requires:
+    cap.core.type.platform:
+      oneOf:
+        - name: kubernetes
+          revision: 0.1.0
+
+  action:
+    runnerInterface: argo.run
+    args:
+      workflow:
+        entrypoint: postgres-install
+        templates:
+          - name: postgres-install
+            steps:
+              - - name: helm-install
+                  # Argo renderer mutates the container definition to ensure that it's created on a given target cluster. 
+                  # It's done based on injected `cap.core.type.platform.kubernetes` TypeInstance
+                  capact-action: helm.install
+```
+
+If necessary we can introduce the `capact-target` property which can be used to mark a given step to be executed on a given cluster. Without `capact-target` step is not mutated, so it's executed where Argo was installed:
+
+```yaml
+# ...
+  requires:
+    cap.core.type.platform:
+      oneOf:
+        - name: kubernetes
+          alias: k8s
+          revision: 0.1.0
+
+# ...
+            steps:
+              - - name: helm-install
+                  capact-action: helm.install
+                  capact-target: k8s # based on alias in requires section 
+# ...
+```
+
+Capact user or admin can override the default Kubernetes cluster used in Implementation via Policy:
+
+```yaml
+rules:
+  - interface: 
+      path: "cap.interface.database.postgresql.install"
+    oneOf:
+      - implementationConstraints: # In first place, find and use an Implementation which:
+          requires: # has the following Type references defined in the `spec.requires` property:
+            - path: "cap.core.type.platform.kubernetes"
+        inject:
+          requiredTypeInstances: # For such Implementation, inject the following TypeInstance. 
+            - id: 9038dcdc-e959-41c4-a690-d8ebf929ac0c
+              description: "EU workload cluster"
+```
+
+### Consequences
+
+To reduce the boilerplate and support multi-cluster in Capact, the following items need to be resolved:
+
+1. Remove `cap.attribute.containerization.kubernetes.kubeconfig-support` from **attributes**.
+2. Merge the `cap.type.containerization.kubernetes.kubeconfig` to `cap.core.type.platform.kubernets` as we don't have the [Type composition](https://capact.io/docs/feature/type-features#type-composition) yet.
+3. Allow injecting `cap.core.type.platform.kubernets` via Policy. We already use such approach for AWS and GCP credentials.
+4. Add option to use `inject.requiredTypeInstances` via artifact name in Workflow Policy. This will solve https://github.com/capactio/capact/issues/538 as we will do that via Policy instead.
+5. Solve [Support setting relations for optional TypeInstances in workflows](https://github.com/capactio/capact/issues/537) issue, but also take into account TypeInstances from the `requires` section.
```

Comment:
> but also take into account TypeInstances from the `requires` section.

Isn't that duplication of the point No 4 in consequences?

Reply:
discussed f2f, invalid 👍